Indicators/OptionGreekIndicatorBase.cs

```diff
@@ -42,17 +42,21 @@ public abstract class OptionGreeksIndicatorBase : OptionIndicatorBase
         /// <param name="option">The option to be tracked</param>
         /// <param name="riskFreeRateModel">Risk-free rate model</param>
         /// <param name="dividendYieldModel">Dividend yield model</param>
+        /// <param name="mirrorOption">The mirror option for parity calculation</param>
         /// <param name="optionModel">The option pricing model used to estimate the Greek</param>
         /// <param name="ivModel">The option pricing model used to estimate IV</param>
         /// <param name="period">The lookback period of historical volatility</param>
         protected OptionGreeksIndicatorBase(string name, Symbol option, IRiskFreeInterestRateModel riskFreeRateModel, IDividendYieldModel dividendYieldModel,
-            OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, OptionPricingModelType? ivModel = null, int period = 2)
+            Symbol mirrorOption = null, OptionPricingModelType optionModel = OptionPricingModelType.BlackScholes, OptionPricingModelType? ivModel = null, int period = 2)
             : base(name, option, riskFreeRateModel, dividendYieldModel, optionModel, period)
         {
             ivModel = ivModel ?? optionModel;
-            ImpliedVolatility = new ImpliedVolatility(name + "_IV", option, riskFreeRateModel, dividendYieldModel, (OptionPricingModelType)ivModel, period);
-            
             WarmUpPeriod = period;
+            ImpliedVolatility = new ImpliedVolatility(name + "_IV", option, riskFreeRateModel, dividendYieldModel, mirrorOption, (OptionPricingModelType)ivModel, period);
+            if (mirrorOption != null)
+            {
+                SetMirrorOptionContract(mirrorOption);
+            }
```

Comment:
I think we should remove the `SetMirrorOptionContract` and just handle it in the `base` call passing through the mirror option value we have?